tests/integration_tests/tests/agentless_tests/test_blueprint_metadata.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+
+from integration_tests import AgentlessTestCase
+from integration_tests.tests.utils import get_resource as resource
+
+pytestmark = pytest.mark.group_deployments
+
+
+class BlueprintSetIconTest(AgentlessTestCase):
+    def test_blueprint_upload_and_remove_icon(self):
+        blueprint_id = 'bp'
+        self.client.blueprints.upload(
+            resource('dsl/empty_blueprint.yaml'),
+            entity_id=blueprint_id
+        )
+        self.client.blueprints.upload_icon(
+            blueprint_id,
+            resource('assets/cloudify.png')
+        )
+        self.client.blueprints.remove_icon(blueprint_id)
```

Comment:
maybe we can have some assert between them?

Reply:
There is no good solution to assert icon existence (yet).  I am keeping that in mind though and will get back to it soon.